smdebug/tensorflow/base_hook.py

```diff
@@ -306,12 +306,12 @@ def _get_writers(self, tensor_name, tensor_ref, shape_writers=False) -> List[Fil
                     else:
                         if shape_writers is False:
                             return [self.writer_map[self.device_map[self.chief_worker]]]
-                        else:
```

Comment:
Why do we need a different shape writer?
If `self._saving_shapes_in_step()` is True for every step, do we end up using only shape_writer?
Does it not make sense then to have a single writer and move the is_saving_shape logic to the Writer class?